### PR TITLE
Acceptance test

### PR DESCRIPTION
**Pull request by marzann**
- Added acceptance test for `facebook_data_analyzer`. 
- The tests now include: unit tests, acceptance test, and continuous integration.
- This PR completes the initial test suite. 
- Tests can be updated further in the future as needed.

### CODE REVIEW

**Comment by marzann** [comment]
@Lackoftactics: did you have a chance to review the acceptance test?

File: lib/facebook_data_analyzer/analyzeables/messages.rb [hunk 1]
```diff
@@ -30,7 +30,7 @@ def initialize(catalog:, options: {})
       @file_pattern = '*.html'
       @messages = []
 
-      super(parallel: options.fetch(:parallel))
+      super(parallel: options.fetch(:parallel) { false })
```
Comment (thnukid) [inline lib/facebook_data_analyzer/analyzeables/messages.rb]: I would prefer `options.fetch(:parallel, false)`

File: lib/facebook_data_analyzer.rb [hunk 2]
```diff
@@ -18,10 +18,11 @@
 
 module FacebookDataAnalyzer
   def self.run(options = {})
+    file_output    = 'facebook_data_analyzer'
     catalog        = options.fetch(:catalog)
-    xlsx           = [options.fetch(:filename), 'xlsx'].join('.')
-    html           = [options.fetch(:filename), 'html'].join('.')
-    parallel_usage = options.fetch(:parallel)
+    xlsx           = [options.fetch(:filename) {file_output}, 'xlsx'].join('.')
```
Comment (thnukid) [inline lib/facebook_data_analyzer.rb]: I would prefer 

`[options.fetch(:filename, file_output), 'xlsx'].join('.')`

File: lib/facebook_data_analyzer.rb [hunk 3]
```diff
@@ -18,10 +18,11 @@
...
+    html           = [options.fetch(:filename) {file_output}, 'html'].join('.')
```
Comment (thnukid) [inline lib/facebook_data_analyzer.rb]: same as above

File: lib/facebook_data_analyzer.rb [hunk 4]
```diff
@@ -18,10 +18,11 @@
...
+    parallel_usage = options.fetch(:parallel) { false }
```
Comment (thnukid) [inline lib/facebook_data_analyzer.rb]: same as above

File: spec/lib/facebook_data_analyzer_spec.rb [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+  before(:all) { described_class.run(catalog: test_catalog) }
+
+  describe  '.run' do
+    context '.xlsx' do
+      let!(:workbook) { Workbook::Book.open 'facebook_data_analyzer.xlsx' }
+
+      it 'generates "Friends ranking" worksheet correctly' do
+        sheet = workbook[0]
+        table = sheet.table
+        expect(table["B3"].value).to eq("Suzanne Nash")
```
Comment (thnukid) [inline spec/lib/facebook_data_analyzer_spec.rb]: Use single quotes `'some text'`. Use double quotes when interpolating strings `"#{my_value} some text"` (rubocop would complain)

eg.
` expect(table["B3"].value).to eq("Suzanne Nash")` ->  `expect(table['B3'].value).to eq('Suzanne Nash')`

**Comment by thnukid** [comment]
Besides my suggestions, this is good work and should be merged.

**Comment by marzann** [comment]
@thnukid: Done. :) Thank you for your review!

File: lib/facebook_data_analyzer.rb [hunk 6]
```diff
@@ -18,10 +18,14 @@
 
 module FacebookDataAnalyzer
   def self.run(options = {})
+    file_output    = 'facebook_data_analyzer'
     catalog        = options.fetch(:catalog)
-    xlsx           = [options.fetch(:filename), 'xlsx'].join('.')
-    html           = [options.fetch(:filename), 'html'].join('.')
-    parallel_usage = options.fetch(:parallel)
+    # xlsx           = [options.fetch(:filename) {file_output}, 'xlsx'].join('.')
```
Comment (thnukid) [inline lib/facebook_data_analyzer.rb]: The comments can be removed :)

Comment (marzann) [reply lib/facebook_data_analyzer.rb]: Yes, of course! I meant to do that. :)

**Pull request merged**